sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -3709,6 +3709,18 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val LEGACY_LPAD_RPAD_BINARY_TYPE_AS_STRING =
+    buildConf("spark.sql.legacy.lpadRpadBinaryTypeAsString")
```

Comment:
how about `spark.sql.legacy.lpadRpadAlwaysReturnString`? If true, it's the legacy behavior and the pad function always return string even for binary inputs. If false, the function returns binary if the first argument and the padding pattern parameter are both binary type.

Reply:
Sure! This sounds better. Naming is hard ..